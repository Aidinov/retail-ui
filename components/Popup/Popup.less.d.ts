export const popup: string;
export const shadow: string;
