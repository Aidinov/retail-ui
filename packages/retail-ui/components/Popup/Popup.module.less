@show-transition: transform 0.18s cubic-bezier(0.22, 0.61, 0.36, 1), opacity 0.18s cubic-bezier(0.22, 0.61, 0.36, 1);
@hide-transition: opacity 0.15s ease-out;

:local {
  .popup {
    position: absolute;
    min-width: 18px;

    &-ignore-hover {
      pointer-events: none;
    }
  }

  .content {
    overflow: hidden;
  }

  .transition {
    &-enter {
      opacity: 0.01;

      &.transition-enter-top {
        transform: translateY(10px);
      }

      &.transition-enter-bottom {
        transform: translateY(-10px);
      }

      &.transition-enter-left {
        transform: translateX(10px);
      }

      &.transition-enter-right {
        transform: translateX(-10px);
      }
    }

    &-enter-active {
      transition: @show-transition;
      opacity: 1;
      transform: translate(0, 0);
    }

    &-exit {
      opacity: 0.01;
      transition: @hide-transition;
      transform: translate(0, 0);
    }
  }
}
