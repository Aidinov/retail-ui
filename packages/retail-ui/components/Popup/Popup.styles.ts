import { css } from 'emotion';
import { ITheme } from '../../lib/ThemeManager';

const jsStyles = {
  popup(t: ITheme) {
    return css`
      border-radius: ${t.popupBorderRadius};
      border: ${t.popupBorder} ${t.popupBorderColor};

      .rt-ie8 & {
        border: 1px solid ${t.dropdownMenuBorder};
      }
    `;
  },

  content(t: ITheme) {
    return css`
      border-radius: ${t.popupBorderRadius};
    `;
  },

  contentInner(t: ITheme) {
    return css`
      background: ${t.bgDefault};
    `;
  },

  shadow(t: ITheme) {
    return css`
      filter: ${t.popupDropShadow};
      -webkit-filter: ${t.popupDropShadow};

      .rt-ie-any & {
        box-shadow: ${t.popupBoxShadow};
      }
    `;
  },
};

export default jsStyles;
