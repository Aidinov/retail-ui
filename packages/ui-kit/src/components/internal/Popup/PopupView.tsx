import React from 'react';
import styled from '../../../lib/styled-components';
import ZIndex, { ZIndexProps } from '../ZIndex';
import { prop, ifProp } from 'styled-tools';
import { Transition } from '../../Transition';

export type Direction = 'top' | 'bottom' | 'left' | 'right';
export type TransitionState = 'entering' | 'entered' | 'exiting' | 'exited' | 'unmounted';

export interface PopupViewProps {
  transitionState?: TransitionState;
  direction?: Direction;
  hasShadow?: boolean;
  backgroundColor?: React.CSSProperties['backgroundColor'];
}

class PopupView extends React.Component<PopupViewProps & ZIndexProps> {
  public render() {
    const {
      children,
      transitionState,
      direction,
      hasShadow,
      backgroundColor,
      ...rest
    } = this.props;

    return <ZIndex {...rest}>{children}</ZIndex>;
  }
}

export const PopupStyledView = styled(PopupView)`
  min-width: ${prop('minWidth', '100px')};
  max-width: ${prop('maxWidth', '500px')};
  width: 100%;
  border-radius: 2px;
  background: ${prop('backgroundColor', '#fff')}
    ${ifProp(
      'hasShadow',
      `
    filter: drop-shadow(0 0 1px rgba(0, 0, 0, 0.1)) drop-shadow(0 1px 2px rgba(0, 0, 0, 0.3));
    -webkit-filter: drop-shadow(0 0 1px rgba(0, 0, 0, 0.1)) drop-shadow(0 1px 2px rgba(0, 0, 0, 0.3));
  `
    )};
`;

export const PopupTransition = styled(Transition)`
  position: absolute;
`;
