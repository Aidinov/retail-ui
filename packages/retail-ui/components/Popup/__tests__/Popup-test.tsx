import * as React from 'react';
import { mount } from 'enzyme';
import Popup from '../';
import toJson from 'enzyme-to-json';
import { PopupProps, PopupState } from '../Popup';

describe('Popup', () => {
  it('open and close without crash', () => {
    const anchor = document.createElement('button');

    anchor.id = 'test-id';
    anchor.innerHTML = 'test';

    const wrapper = mount(
      <Popup
        positions={['bottom left', 'bottom right', 'top left', 'top right']}
        opened={false}
        anchorElement={anchor as HTMLElement}
      >
        Test content
      </Popup>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({ opened: true });

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({ opened: false });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('сохраняет свое положение при повторном открытии', () => {
  //   const anchor = document.createElement('button');

  //   anchor.id = 'test-id';
  //   anchor.innerHTML = 'test';

  //   const wrapper = mount<PopupProps, PopupState>((
  //     <Popup
  //       positions={['top left', 'top right', 'bottom left', 'bottom right']}
  //       opened={false}
  //       anchorElement={anchor as HTMLElement}
  //     >
  //       Test content
  //     </Popup>
  //   ) as React.ReactElement<PopupProps>);

  //   expect(wrapper.state('location')).toBeNull();

  //   wrapper.setProps({ opened: true });

  //   expect(wrapper.state('location')).toEqual('bottom left');
  // });
});
