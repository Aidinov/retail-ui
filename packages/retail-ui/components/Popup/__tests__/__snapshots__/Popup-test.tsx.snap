// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup snapshots:  open and close popup: closed popup 1`] = `
<Popup
  anchorElement={
    <button
      id="test-id"
    >
      test
    </button>
  }
  disableAnimations={false}
  hasPin={false}
  hasShadow={false}
  margin={10}
  maxWidth={500}
  opened={false}
  pinOffset={16}
  pinSize={8}
  popupOffset={0}
  positions={
    Array [
      "bottom left",
      "bottom right",
      "top left",
      "top right",
    ]
  }
  useWrapper={false}
>
  <RenderContainer
    anchor={null}
  />
</Popup>
`;

exports[`Popup snapshots:  open and close popup: closed popup again 1`] = `
<Popup
  anchorElement={
    <button
      id="test-id"
    >
      test
    </button>
  }
  disableAnimations={false}
  hasPin={false}
  hasShadow={false}
  margin={10}
  maxWidth={500}
  opened={false}
  pinOffset={16}
  pinSize={8}
  popupOffset={0}
  positions={
    Array [
      "bottom left",
      "bottom right",
      "top left",
      "top right",
    ]
  }
  useWrapper={false}
>
  <RenderContainer
    anchor={null}
  />
</Popup>
`;

exports[`Popup snapshots:  open and close popup: opened popup 1`] = `
<Popup
  anchorElement={
    <button
      id="test-id"
    >
      test
    </button>
  }
  disableAnimations={false}
  hasPin={false}
  hasShadow={false}
  margin={10}
  maxWidth={500}
  opened={true}
  pinOffset={16}
  pinSize={8}
  popupOffset={0}
  positions={
    Array [
      "bottom left",
      "bottom right",
      "top left",
      "top right",
    ]
  }
  useWrapper={false}
>
  <RenderContainer
    anchor={null}
  >
    <Transition
      appear={true}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 0,
          "exit": 200,
        }
      }
      unmountOnExit={true}
    >
      <ZIndex
        className="popup transition-enter-top"
        delta={1000}
        key="dummy"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        render={true}
        style={
          Object {
            "left": -9999,
            "maxWidth": 500,
            "top": -9999,
          }
        }
      >
        <div
          className="popup transition-enter-top"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "left": -9999,
              "maxWidth": 500,
              "top": -9999,
              "zIndex": 1000,
            }
          }
        >
          <div
            className="content"
          >
            <div
              className="contentInner"
              style={
                Object {
                  "backgroundColor": undefined,
                }
              }
            >
              Test content
            </div>
          </div>
        </div>
      </ZIndex>
    </Transition>
    <Portal
      key="portal-ref"
      rt_rootID={1}
    >
      <noscript
        data-render-container-id={1}
      />
    </Portal>
  </RenderContainer>
</Popup>
`;
